src/content/app/genome-browser/components/drawer/drawer-views/variant-summary/variant-vcf/VariantVCF.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+/**
+ * NOTE:
+ * - This component almost certainly will be reused enough to become a shared component
+ * - This component will need to learn how to display VCF string for multiple long alleles.
+ *   See example:
+ *   - https://xd.adobe.com/view/b485e6c9-546a-4627-a5a6-0edf77651f09-e1d6/screen/973f8bb3-05c1-4cf8-9220-08009640c571?fullscreen
+ * - Should the Copy component be part of VariantVCF component; or should VariantVCF component
+ *   expose a method that a parent can use to get the VCF string from VariantVCF? The second
+ *   options seems more composable.
+ */
+
+type MinimumVariantData = {
+  name: string;
+  alleles: {
+    reference_sequence: string;
+    allele_sequence: string;
+    slice: {
+      location: {
+        start: number;
+      };
+      region: {
+        name: string;
+      };
+    };
+  }[];
+};
+
+type PropsWithVariant = {
+  variant: MinimumVariantData;
+};
+
+type PropsWithSequence = {
+  vcfSequence: string;
+};
+
+type Props = PropsWithVariant | PropsWithSequence;
+
+const VariantVCF = (props: Props) => {
+  const vcfSequence = arePropsWithVariant(props)
```

Comment:
> can't we just check if variant in props here

Typescript won't allow that. The props here are a union of two types: one has a `variant` property, and the other doesn't. Typescript won't let you access a property that does not exist on all members of the union. See [example]([url](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBvAsAKClUkBcUA5AGYJEDcWOZcAypAMYCWAhgDaFECSUcAduxBQIAD2YBnYH36xyVbFEZwAtioFderFVFZLV62WABOcSMdCUsAXyxY80AEKtjyNNVzgIXAEYurin7GDBAsHJoyQiLiUjJQzsYBOMpqGsRaOnophlAmZhAWIAG2mPZeUAAKpmASbvCIAD7xLgpYyvyxwBBSAIxuABR5NYRVZhIAlMgAfO6K7bGonpAANPqpstZuQxIUUAD0e1AA8gDSNnaY89JdUgBMA9sj1RPTs8kCC0sQq9kCq7QhMLsKCbFDbXYHKAAOSOABVjmdMCU2h9rt1gABmB7PJ5jSZIGYYOaotBfEFbZ6tRTMEhQfoOZBIFCkciTIk4d4dODsCAAOnYcAA5oNnryAUw2OxxhDDqcPCVrBQgA)) in typescript playground.

What one could do is mark the mutually exclusive properties with type `never`, like so ([playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBvAsAKClUkBcUA5AGYJEDcWOZcAypAMYCWAhgDaFECSUcAduxBQIAD2YBnYH36xyVbFABGrAE4MILDoX4QAbhFUKcjOAFszArr1ZmorKKYsCoYVXEirQlLAF8sWHjQAEJqyGjUuOAQXCqqPopxGlqcxLwCQiLiUjJQofHGULTJbKm6BkaRTpb81vZ2DtUubh6G3gr+mIHRUAAK7mAS4fCIAD55agpYpvw5wBBSAIzhABQtg4T9HhIAlMgAfBGKMzmoUZAANI7mNVC+4esSFFAA9C9QAPIA0n4BmCfSeZSABMq0emwGuwORxMAlO5wgVya-CuxSYpTuD0hzzeUAAch8ACqfH6YTrTOGAhbAADMYMhEO2eyQhwwx0paARmJQjymimYJCgKyCyCQKFI5D2bJwsNmcHYEAAdOw4ABzNaQxVozSlHY497fSKdXwUIA)). I don't have a strong feeling about whether this would be preferable here.